### DISCUSSION TITLE
Testnet connection information inconsistency

### DISCUSSION THREAD

**Discussion by tgrosinger** [markqvist/Reticulum#83] [Help & Questions] [answered]
I'm having some trouble getting started with NomadNet on the Testnet servers.

In the guide, it lists the test network as 

```
  [[RNS Testnet Frankfurt]]
    type = TCPClientInterface
    interface_enabled = yes
    outgoing = True
    target_host = frankfurt.rns.unsigned.io
    target_port = 4965
```

However in the [Reticulum Docs](https://markqvist.github.io/Reticulum/manual/gettingstartedfast.html#connect-to-the-public-testnet) the Frankfurt section is actually titled Dublin and has a different port than above. Also both the Dublin and Frankfurt sections use a different target host domain.

What is the correct connection information to use? I was able to connect with `telnet frankfurt.rns.unsigned.io 5377` so that is what I am using right now in nomadnet, but no matter what combination I try I have an empty announce log and haven't been able to connect to anyone.

Thank you!

**Update**, I set my configuration to this and left the program up for a half hour.

```
  [[RNS Testnet Frankfurt]]
    type = TCPClientInterface
    interface_enabled = yes
    outgoing = True
    target_host = frankfurt.connect.reticulum.network
    target_port = 5377

  [[RNS Testnet Dublin]]
    type = TCPClientInterface
    enabled = yes
    target_host = dublin.connect.reticulum.network
    target_port = 4965
```

I now see 4 nodes and 1 peer in my announce stream. I was able to send a message to Echo Test and received an echo as I expected. However I do not see the destination mentioned at the end of [this blog post](https://unsigned.io/rnode-a-new-model/). Do I need to trust one of these nodes in order to see more peers? How do I know which of those nodes are the Testnet nodes themselves?

Apologies for the entry level questions. I've read most of the docs I can find, but it's still a lot to absorb and put into practice.

**Comment by Swissbandit** [comment]
Hi there,the peers and nodes will announce all,from time to time,the longer you are on the network,the more announces youll see

**Reply by tgrosinger** [reply]
Ah I found the Frankfurt node now and the unsigned node. Took a surprisingly long time for them to appear.

**Reply by markqvist** [reply]
They only announce at pretty low frequencies, so you will have to stay connected for some time for them to appear. Most of the nodes will announce only every six or twelve hours.

**Reply by tgrosinger** [reply]
Hmm, this is very interesting. Is that just to preserve bandwidth?

This makes the start-up experience for new users a little difficult for some use cases. For example, say I want to read the pages on RNS Frankfurt. Is there a way I could have found that sooner rather than waiting for it to announce?

**Reply by Swissbandit** [reply]
Thats why you can save known nodes.
And to bring beginners in an adressbook can be made and shared
As a beginners index.mu page

**Reply by markqvist** [reply]
Yes, it is primarily just to not spam the network with a lot of unneccessary announces. You could always connect directly to the RNS Frankfurt node, by simply going to the browser, pressing Ctrl+U to bring up the URL dialog, and then pasting the RNS Frankfurt node address `5cda089fc42675bdd904c6d06db87d49` into the URL field, and pressing `< Go >`. Even if your nomadnet client had never heard about the node or address, it would try to resolve and connect to it.

I guess this is not very clear, and could be made a lot more user friendly by including some well-known "demo node" addresses somewhere in the docs. Would love to hear your thoughts on what the best way to go about this could be.

**Reply by tgrosinger** [reply]
I think including that destination address in a comment right next to the interface config in the docs would be perfect. I had the interface config, but just wasn't sure what the address of the node I was looking for was.

**Comment by Swissbandit** [comment]
There is an old testnet,but here are the new coordinates,for the updated versions of rns

frankfurt.connect.reticulum.network
On port 5377

Or the dublin connection with port 4965

**Comment by Swissbandit** [comment]
And the lxmf adress you have can be manually added,without an anounce,but then you ask the network for a pathkey and you may get it.then try to message with propagation,assuming the peer offline therefore not announcing.

**Comment by markqvist** [comment, marked as answer]
Thanks for reporting that inconsistency! Thought I had it cleaned up, but apparently I missed some parts.

All of what @Swissbandit says is correct, I just want to add the definitive references for the current testnet connections here as  well. These are the public entry-points for the current testnet, that is "officially" available: 

```
# TCP/IP interface to the Dublin Hub
  [[RNS Testnet Dublin]]
    type = TCPClientInterface
    enabled = yes
    target_host = dublin.connect.reticulum.network
    target_port = 4965

# TCP/IP interface to the Frankfurt Hub
  [[RNS Testnet Frankfurt]]
    type = TCPClientInterface
    enabled = yes
    target_host = frankfurt.connect.reticulum.network
    target_port = 5377

# Interface to I2P Hub A
  [[RNS Testnet I2P Hub A]]
    type = I2PInterface
    enabled = yes
    peers = mrwqlsioq4hoo2lmeeud7dkfscnm7yxak7dmiyvsrnpfag3z5tsq.b32.i2p

# Interface to I2P Hub B
  [[RNS Testnet I2P Hub B]]
    type = I2PInterface
    enabled = yes
    peers = iwoqtz22dsr73aemwpw7guocplsjjoamyl7sogj33qtcd6ds4mza.b32.i2p
```
Since anyone can expand the testnet, there may be several other publicly available entry-points that can be used, but these at least will get anyone started. You can reach that LXMF address via the testnet, but it is currently not constantly online, since I am on the move, without much connectivity to the Internet ;) You can always leave messages on the propagation nodes, and I will get them when I can.

**Reply by tgrosinger** [reply]
I have marked RNS Frankfurt as trusted and as the default propagation node, however it was still telling me it was looking for a path. Is there documentation about how to leave a message on the propagation nodes that I can review?

Of course as I look right now it seems that you have come online so the error message I was seeing has disappeared 🙂

**Reply by markqvist** [reply]
Both Nomadnet and Sideband should deliver messages via propagation nodes if it is not able to establish a direct link to the destination after a couple of tries (as long as you have at least one trusted node set in Nomadnet).

If the error message you are talking about is the one in the nomadnet conversation view, saying `You cannot currently message this peer, since it's identity keys are not known`, this indicates that your local system has never received an announce from the address, and therefore does not know the cryptograhic keys to establish encrytped communication with it. In nomadnet, you can just ask the network for this info, by selecting the conversation in the list, and pressing Ctrl+E, and then selecting "Query network for keys". If you are conencted to any network that knows about the destination, it will forward an announce to you, and you can then communicate :)

But yes, I did come online now and my client sent out an announce on the network, so you should have gotten it ;)

**Reply by tgrosinger** [reply]
I tried using Ctrl+e a few times, including after I had set the RNS Frankfurt as my default propagation node. There was no visual feedback as to whether it did anything, or any update in the logs. The message you described persisted too.

**Reply by markqvist** [reply]
You need to mark the conversation in the list before pressing Ctrl-E, but I realise this is a bit unintuitive. I will add a button in the missing keys message for querying for it directly, it will be much more intuitive.
